scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHandshaker.java

```diff
@@ -166,7 +166,7 @@ public ClientHandshaker(DTLSSession session, RecordLayer recordLayer, SessionLis
 
 			// we always support receiving a RawPublicKey from the server
 			this.supportedServerCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
-			if (rootCertificates != null && rootCertificates.length > 0) {
+			if (!certificateStore.isEmpty()) {
```

Comment:
I just translate the previous code.
But I doubt this is the good way to do now the store is dynamic.
Maybe we should get  certificates from credentials store only once, then store it in the Handshaker and always use the same.
This sounds better but it will consume more memory if there is lot of Handshakers and a lot of Certificates store in a persistent store without cache.

(if we do that I remove the isEmpty from the store)

Reply:
I'm not sure, how dynamic the store is intended to be.
What should happen, if a trusted certificate is removed during an ongoing handshake?

Reply:
We look if the store is empty to express if we support X509 certificate.
Then later we re-ask to the store the trusted certificates to verify foreign peer certificate.
E.g. :
1. We check the store, it is not empty so we ask for X509 certificate.
2. Later we validate foreign peer certificate,  at this moment  the store is now so we can not validate it.

It's not a big deal,  just a bit strange.

The other points maybe more problematic is about 
https://github.com/eclipse/californium/pull/533/files#diff-5da9149ee37db8bcc492b03743f00dfeR620
This time we will send list of acceptable certificate_authorities then later when we receive the client certificate we could face a situation where the right authorities is not in the store anymore.